### ISSUE TITLE
Collapse selection to end after inserting hyperlink on selected text

### ISSUE THREAD

**Issue by Philipp91** [tinymce/tinymce#6711]
tl;dr: Just read the bold-printed text below.

### What is the current behavior? How to reproduce

* https://fiddle.tiny.cloud/
* Write some text (e.g. "Test") and **select it**.
* Press Ctrl+K to **insert a hyperlink**. Give it an arbitrary URL (eg. `http://test.com`) and click "Save".
* Observe that "Test" is now selected, i.e. this is the selection:
  ```bash
  Sometext |<a href="...">Linktext</a>| after
           |--------------------------|    <---this is where the selection is
  ```
* Typing now removes the link has just been inserted, e.g. if I type `B` I get: `Sometext b| after`.
* In order to continue typing behind the newly inserted link (as one would normally want when typing texts), the user has to press the right arrow key twice (once to collapse the selection to the end resulting in `Sometext <a href="...">Linktext|</a> after`, and one to step outside the link text resulting in `Sometext <a href="...">Linktext</a>| after`).

### What is the expected behavior?

**After inserting a link, the cursor should be placed _behind_ the link**, so that the next typed character is not added to the link or replaces it. I.e.

```bash
Sometext <a href="...">Linktext</a>| after
                                   ^---- this is where the caret should be after inserting the link
```

To me this behavior makes more sense because the user likely wants to continue typing. Given that they just had the opportunity to edit the link text in the dialog, it's unlikely that they want to modify it again in the editor after closing the dialog.

The desired behavior is also the behavior of Google Docs and Microsoft Word (at least their web version, I don't have it installed).

**Does it matter? I think so.** I have muscle memory to paste links on selected text and continue typing behind it (Ctrl+K, Ctrl+V, Enter, Space, ...). It works in all applications I use, except with TinyMCE I end up erasing my inserted links.

### Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?

I'm using the latest Chrome version on Windows and tinymce5-cloud-stable. This also reproduces e.g. in 5.0.0.

**Comment by Philipp91** [comment]
I noticed that the behavior is as expected when inserting a link without selecting any text beforehand (i.e. also typing the link text into the dialog). So I guess TinyMCE's behavior is to always keep the selection as is, which would make this a feature request rather than a bug report, because the selection needs to be actively changed. Still I think it's the right behavior.

**Comment by SimonFc** [comment]
Thank you for the detailed description. This appears to be a regression as the expected behavior you're describing worked before 4.6.0 and I can't see anything in the changelog about this.

**Comment by spocke** [comment]
So this has now been fixed in 7.7.0 so closing this ticket.
